src/Stl.Interception/ServiceCollectionTypeFactoryExt.cs

```diff
@@ -0,0 +1,69 @@
+using Stl.Interception.Interceptors;
+
+namespace Stl.Interception;
+
+public static class ServiceCollectionTypeFactoryExt
+{
+    public static IServiceCollection AddScopedTypeFactory<TFactory>(this IServiceCollection services)
+        where TFactory : class, IRequiresAsyncProxy
+    {
+        return AddTypeFactory<TFactory>(services, ServiceLifetime.Scoped);
+    }
+
+    public static IServiceCollection AddSingletonTypeFactory<TFactory>(this IServiceCollection services)
+        where TFactory : class, IRequiresAsyncProxy
+    {
+        return AddTypeFactory<TFactory>(services, ServiceLifetime.Singleton);
+    }
+
+    public static IServiceCollection AddTypeFactory<TFactory>(this IServiceCollection services, ServiceLifetime lifetime)
+        where TFactory : class, IRequiresAsyncProxy
+    {
+        return AddTypeFactory<TFactory, TypeFactoryInterceptor>(services, lifetime);
+    }
+
+    public static IServiceCollection AddTypeFactory<TFactory, TInterceptor>(this IServiceCollection services, ServiceLifetime lifetime)
+        where TFactory : class, IRequiresAsyncProxy
+        where TInterceptor : Interceptor
+    {
+        return services.AddTypeFactory(typeof(TFactory), typeof(TInterceptor), lifetime);
+    }
+
+    public static IServiceCollection AddTypeFactory(this IServiceCollection services, Type factoryType, Type interceptorType, ServiceLifetime lifetime)
+    {
+        // All implementations converge here.
+        // Single source of Proxies.GetProxyType if not specified by any of the TProxy-variants.
+        return AddTypeFactory(services, factoryType, Proxies.GetProxyType(factoryType), interceptorType, lifetime);
```

Comment:
Didn't want to introduce a bunch of calls to Proxies.GetProxyType, thus here we are.
All options call into this overload at the end, which either throws on Add()-call because no factory was found, or continues and it is guaranteed that there are no missing types on registration